go-controller/pkg/util/ovs.go

```diff
@@ -535,6 +535,24 @@ func RunOVNNorthAppCtl(args ...string) (string, string, error) {
 	return strings.Trim(strings.TrimSpace(stdout.String()), "\""), stderr.String(), err
 }
 
+// RunOVNControllerAppCtl runs an 'ovs-appctl -t ovn-controller command'.
+func RunOVNControllerAppCtl(args ...string) (string, string, error) {
+	var cmdArgs []string
+
+	pid, err := ioutil.ReadFile(runner.ovnRunDir + "ovn-controller.pid")
```

Comment:
you shouldn't have to do this... you should be able to just say `ovn-appctl -t ovn-controller` and it will read the pid file and then open the correct socket file based on that.

um... yes, this means all of the existing appctl wrappers in this file are doing pointless extra work...

Reply:
@danwinship you cannot just use `ovn-appctl -t ovn-controller` as it will not work. The containers within the pod are not sharing the PID namespace. So, if you try to do what you say, we will get the following error

```
$ ovn-appctl -t ovn-controller list-commands
2020-05-06T20:42:07Z|00001|daemon_unix|WARN|/var/run/ovn/ovn-controller.pid: stale pidfile for pid 60
 being deleted by pid 0
ovn-appctl: cannot read pidfile "/var/run/ovn/ovn-controller.pid" (No such process)
```
```

Reply:
Oh. Right, this is basically the same thing as https://bugzilla.redhat.com/show_bug.cgi?id=1803920.

If you look at [ovn-appctl.c](https://github.com/ovn-org/ovn/blob/master/utilities/ovn-appctl.c#L203), it looks like it's doing exactly the same thing this golang code is doing. Except that it's not, because `read_pidfile` doesn't just read the pidfile, it tries to be "clever" and breaks things. :frowning_face: